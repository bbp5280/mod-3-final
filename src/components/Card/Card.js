import React from 'react';

const Card = ( {name,
  founded,
  seats,
  titles,
  coatOfArms,
  ancestralWeapons,
  words}) => {

  const buildListItems = (houseFeatures, title) => {
    return houseFeatures.map(feature =>{
      return <li key={feature}> {title}: {feature}</li>;
    });
  };

  const checkForValue = (houseFeature) => {
    return  houseFeature === '' ? 'NA': houseFeature;
  };

  return (
    <div className='card'>
      <div className='inner-card'>
        <h2>{name}</h2>
        <h3>Founded: {checkForValue(founded)}</h3>
        <ul>
          {buildListItems(seats, 'Seats')}
        </ul>
        <ul>
          {buildListItems(titles, 'Titles')}
        </ul>
        <p>Coat Of Arms: {coatOfArms}</p>
        <ul>
          {buildListItems(ancestralWeapons, 'Ancestral Weapons')}
        </ul>
        <p>Words: {checkForValue(words)}</p>
      </div>
    </div>
  );
};

export default Card;
