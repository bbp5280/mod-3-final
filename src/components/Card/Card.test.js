import { shallow, configure } from 'enzyme';
import Card from './Card';
import React from 'react';
import  Adapter from 'enzyme-adapter-react-15';

configure({ adapter: new Adapter() });

describe('CardCatalog snapshot', () => {

  it('should always match the snapshot', () => {
    const wrapper = shallow(<Card name='someName'
      founded='someTime'
      seats={['seat', 'seat 2']}
      titles={['title', 'title2']}
      coatOfArms='someCoat'
      ancestralWeapons={['bar']}
      houseWords='word'/>);

    expect(wrapper).toMatchSnapshot();

  });
});
