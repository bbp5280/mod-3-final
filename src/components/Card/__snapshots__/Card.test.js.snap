// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardCatalog snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    ancestralWeapons={
        Array [
            "bar",
          ]
    }
    coatOfArms="someCoat"
    founded="someTime"
    houseWords="word"
    name="someName"
    seats={
        Array [
            "seat",
            "seat 2",
          ]
    }
    titles={
        Array [
            "title",
            "title2",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="inner-card"
>
        <h2>
                someName
        </h2>
        <h3>
                Founded: 
                someTime
        </h3>
        <ul>
                <li>
                         
                        Seats
                        : 
                        seat
                </li>
                <li>
                         
                        Seats
                        : 
                        seat 2
                </li>
        </ul>
        <ul>
                <li>
                         
                        Titles
                        : 
                        title
                </li>
                <li>
                         
                        Titles
                        : 
                        title2
                </li>
        </ul>
        <p>
                Coat Of Arms: 
                someCoat
        </p>
        <ul>
                <li>
                         
                        Ancestral Weapons
                        : 
                        bar
                </li>
        </ul>
        <p>
                Words: 
                word
        </p>
</div>,
      "className": "Card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            someName
</h2>,
          <h3>
            Founded: 
            someTime
</h3>,
          <ul>
            <li>
                         
                        Seats
                        : 
                        seat
            </li>
            <li>
                         
                        Seats
                        : 
                        seat 2
            </li>
</ul>,
          <ul>
            <li>
                         
                        Titles
                        : 
                        title
            </li>
            <li>
                         
                        Titles
                        : 
                        title2
            </li>
</ul>,
          <p>
            Coat Of Arms: 
            someCoat
</p>,
          <ul>
            <li>
                         
                        Ancestral Weapons
                        : 
                        bar
            </li>
</ul>,
          <p>
            Words: 
            word
</p>,
        ],
        "className": "inner-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "someName",
          },
          "ref": null,
          "rendered": "someName",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Founded: ",
              "someTime",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Founded: ",
            "someTime",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                 
                Seats
                : 
                seat
</li>,
              <li>
                 
                Seats
                : 
                seat 2
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "seat",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "Seats",
                  ": ",
                  "seat",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "Seats",
                ": ",
                "seat",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "seat 2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "Seats",
                  ": ",
                  "seat 2",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "Seats",
                ": ",
                "seat 2",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                 
                Titles
                : 
                title
</li>,
              <li>
                 
                Titles
                : 
                title2
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "title",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "Titles",
                  ": ",
                  "title",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "Titles",
                ": ",
                "title",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "title2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "Titles",
                  ": ",
                  "title2",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "Titles",
                ": ",
                "title2",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Coat Of Arms: ",
              "someCoat",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Coat Of Arms: ",
            "someCoat",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                 
                Ancestral Weapons
                : 
                bar
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "bar",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "Ancestral Weapons",
                  ": ",
                  "bar",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "Ancestral Weapons",
                ": ",
                "bar",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Words: ",
              "word",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Words: ",
            "word",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="inner-card"
>
          <h2>
                    someName
          </h2>
          <h3>
                    Founded: 
                    someTime
          </h3>
          <ul>
                    <li>
                               
                              Seats
                              : 
                              seat
                    </li>
                    <li>
                               
                              Seats
                              : 
                              seat 2
                    </li>
          </ul>
          <ul>
                    <li>
                               
                              Titles
                              : 
                              title
                    </li>
                    <li>
                               
                              Titles
                              : 
                              title2
                    </li>
          </ul>
          <p>
                    Coat Of Arms: 
                    someCoat
          </p>
          <ul>
                    <li>
                               
                              Ancestral Weapons
                              : 
                              bar
                    </li>
          </ul>
          <p>
                    Words: 
                    word
          </p>
</div>,
        "className": "Card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              someName
</h2>,
            <h3>
              Founded: 
              someTime
</h3>,
            <ul>
              <li>
                             
                            Seats
                            : 
                            seat
              </li>
              <li>
                             
                            Seats
                            : 
                            seat 2
              </li>
</ul>,
            <ul>
              <li>
                             
                            Titles
                            : 
                            title
              </li>
              <li>
                             
                            Titles
                            : 
                            title2
              </li>
</ul>,
            <p>
              Coat Of Arms: 
              someCoat
</p>,
            <ul>
              <li>
                             
                            Ancestral Weapons
                            : 
                            bar
              </li>
</ul>,
            <p>
              Words: 
              word
</p>,
          ],
          "className": "inner-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "someName",
            },
            "ref": null,
            "rendered": "someName",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Founded: ",
                "someTime",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Founded: ",
              "someTime",
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                   
                  Seats
                  : 
                  seat
</li>,
                <li>
                   
                  Seats
                  : 
                  seat 2
</li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "seat",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "Seats",
                    ": ",
                    "seat",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "Seats",
                  ": ",
                  "seat",
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "seat 2",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "Seats",
                    ": ",
                    "seat 2",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "Seats",
                  ": ",
                  "seat 2",
                ],
                "type": "li",
              },
            ],
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                   
                  Titles
                  : 
                  title
</li>,
                <li>
                   
                  Titles
                  : 
                  title2
</li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "title",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "Titles",
                    ": ",
                    "title",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "Titles",
                  ": ",
                  "title",
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "title2",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "Titles",
                    ": ",
                    "title2",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "Titles",
                  ": ",
                  "title2",
                ],
                "type": "li",
              },
            ],
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Coat Of Arms: ",
                "someCoat",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Coat Of Arms: ",
              "someCoat",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                   
                  Ancestral Weapons
                  : 
                  bar
</li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "bar",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "Ancestral Weapons",
                    ": ",
                    "bar",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "Ancestral Weapons",
                  ": ",
                  "bar",
                ],
                "type": "li",
              },
            ],
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Words: ",
                "word",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Words: ",
              "word",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
